packages/dart_frog_auth/lib/src/dart_frog_auth.dart

```diff
@@ -0,0 +1,90 @@
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:dart_frog/dart_frog.dart';
+
+extension on Map<String, String> {
+  String? authorization(String type) {
+    final value = this['Authorization']?.split(' ');
+
+    if (value != null && value.length == 2 && value.first == type) {
+      return value.last;
+    }
+
+    return null;
+  }
+
+  String? bearer() => authorization('Bearer');
+  String? basic() => authorization('Basic');
+}
+
+/// {@template basic_auth}
+/// Authentication that uses the `Authorization` header with the `Basic` scheme.
+///
+/// `Basic` scheme expects the header to be in the format:
+/// ```
+/// Authorization: Basic <token>
+/// ```
+///
+/// Token should be a base64 encoded string of the format:
+/// ```
+/// <username>:<password>
+/// ```
+///
+/// In order to use this middleware, you must provide a function that will
+/// return a user object from the username and password.
+///
+/// If the given function returns null for the given username and password,
+/// the middleware will return a `401 Unauthorized` response.
+/// {@endtemplate}
+/// Builds the middleware to be used on the dart frog server.
```

Comment:
Is excluding the last line from the template on purpose? Is the last line still needed?

```suggestion
/// {@endtemplate}
/// Builds the middleware to be used on the dart frog server.
```